src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala

```diff
@@ -24,8 +30,26 @@ class LogReceiver extends Server  {
     get {
       log.debug("get /components executed")
       complete("dummy response")
+    } ~ path(Segment) { componentId =>
+      post {
+        // Handle POST of an existing component
+        entity(as[String]) { logLine =>
+          log.debug(s"Received POST on /components/$componentId with log line $logLine")
+          complete(handlePost(logReceiverActor ? SubmitLog(componentId, logLine)))
+        }
+      }
     }
   }
 
   override def routes: Route = componentsRoute
+
+  private def handlePost(resultFuture: Future[Any]): Future[HttpResponse] = {
+    resultFuture.flatMap {
+      case LogSubmitted(componentId, logLine) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462588214/issues/source?bid=3220915&fileBranchId=3255667#l51)